upload problem 2583 baekjoon(minwook)

YOONMINWOOK opened the pull request:
use dfs, loop

YOONMINWOOK committed:
upload problem 2583 baekjoon

boohyunsik commented on MINWOOK/2583-윤민욱.java:
```diff
@@ -0,0 +1,100 @@
+			int j = sc.nextInt();
+			int x = sc.nextInt();
+			int y = sc.nextInt();
+			dfs(j, i, y, x, isVisited);
```
매개변수로 2차원 배열(여기서는 boolean 배열이라 상관없을수도 있지만)넘기면 `메모리를 금방 잡아먹기 떄문에` 추천되는 방법은 아니라고 생각함.
차라리 `전역 변수로 선언`하고 `초기화를 해주는게` 더 좋을듯

boohyunsik commented on MINWOOK/2583-윤민욱.java:
```diff
@@ -0,0 +1,100 @@
+		}
+		if (j + 1 < y && !isVisited[i][j + 1]) {
+
+			dfs(i, j + 1, x, y, isVisited);
```
<b>isVisited를 전역으로 사용한다면</b>
```java
isVisited[i][j] = true;
dfs(i, j+1, x, y);
isVisited[i][j] = false;
```
이런식으로 코딩 가능할듯

boohyunsik commented on MINWOOK/2583-윤민욱.java:
```diff
@@ -0,0 +1,100 @@
+		if (y - 1 >= 0 && coordinate[x][y - 1] != BLOCKED && !isVisited[x][y - 1]) {
+
+			solution(x, y - 1);
+		}
```
요건 오늘 설명해준
```java
int[] dx = {1, 0, -1, 0};
int[] dy = {0, 1, 0, -1};
```
이런 배열을 활용하면 훨씬 깔끔하게 코딩 가능
```java
for (int i=0; i<4; i++) {
   if (enable(x + dx[i], y + dy[i]) && coordinate[x + dx[i]][y + dy[i]] != BLOCKED && !isVisited[x + dx[i]][y + dy[i]]) {
      solution(x + dx[i], y + dy[i]);
   }
}
```

YOONMINWOOK committed:
upload 15683backjoon
i used CCTV class which has its directions array, rotation function, and inspection function.
i used common dfs to solve this problem.
